one_fm/one_fm/report/issue_type/issue_type.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) 2013, omar jaber and contributors
+# For license information, please see license.txt
+
+from warnings import filters
+import frappe
+from frappe import _
+
+def execute(filters=None):
+	columns, data = get_columns(filters), get_data(filters)
+	return columns, data
+
+def get_columns(filters):
+	return [
+		_("Department") + ":Link/Department:250",
+		_("Issue Type") + ":Data:250",
+		_("Custom Issue Type") + ":Data:250",
+		_("Count") + ":Int:250"
+	]
+
+def get_conditions(filters=None):
+	conditions = {}
+	if filters:
+		if filters.get("department"):
+			conditions.update({'department': filters.get("department")})
+		
+		if filters.get("issue_type"):
+			conditions.update({'issue_type': filters.get("issue_type")})
+
+	return conditions
+
+def get_data(filters=None):
+	conditions = get_conditions(filters)
+	result = []
+
+	query_res = frappe.db.sql("""
+		SELECT department, issue_type, your_issue_type, count(*) from `tabIssue`
+		WHERE issue_type = %(issue_type)s AND department = %(department)s
+		GROUP BY department, your_issue_type
+		HAVING count(*) > 0
+	""", {'issue_type': conditions.get("issue_type"), 'department': conditions.get("department")}, as_dict=1)
+
+	for record in query_res:
+		row = [
+			record["department"],
+			record["issue_type"],
+			record["your_issue_type"] if record["your_issue_type"] else "",
+			record["count(*)"]
+		]
+
+		result.append(row)
+	
+	return result
+
+@frappe.whitelist()
+def get_filtered_issue_types(doctype, txt, searchfield, start, page_len, filters):
+	res = []
+	dept = frappe.get_doc("Department", filters.get("department"))
+
+	for d in dept.issue_types:
+		res.append([d.issue_type])
+
+	return res
```

Comment:
Instead of this you can make use of the method in `utils.py` https://github.com/ONE-F-M/One-FM/blob/2870d945a6816a9feae6d67368b341c7cf5c47dc/one_fm/utils.py#L2327 in staging branch

Reply:
@pjamsheer Yeah that would be better. Thanks!

Reply:
Anytime